src/monster.h

```diff
@@ -411,11 +411,12 @@ class monster : public Creature
                                     const std::string &npc_msg ) const override;
         void add_msg_player_or_npc( game_message_type type, const std::string &player_msg,
                                     const std::string &npc_msg ) const override;
-
         // TEMP VALUES
         tripoint wander_pos; // Wander destination - Just try to move in that direction
         int wandf;           // Urge to wander - Increased by sound, decrements each move
         std::vector<item> inv; // Inventory
+        player *mounted_player; // player that is mounting this creature
```

Comment:
Please, at least initialize this to null instead of leaving it uninitialized. Currently using this member is UB.
```suggestion
        player *mounted_player = nullptr;
```

Reply:
Btw. what happens when I give a riding NPC a task in a base camp? The NPC vanishes from the map, but what about its mount? And what when the NPC comes back and the mount is somewhere else? Etc.

Reply:
I'll look into that, the easiest thing to do would be to make them dismount before going off on a companion mission, until I find a better solution.